ffs/filcold/filcold.go

```diff
@@ -219,22 +229,13 @@ func (fc *FilCold) waitForDeals(ctx context.Context, c cid.Cid, storeResults []d
 			continue
 		}
 		if di.StateID == storagemarket.StorageDealActive {
-			var d *deals.StoreResult
-			for _, dr := range storeResults {
-				if dr.ProposalCid == di.ProposalCid {
-					d = &dr
-					break
-				}
-			}
-			if d == nil {
-				return nil, errors, fmt.Errorf("deal watcher return unasked proposal, this must never happen")
-			}
 			activeProposals[di.ProposalCid] = &ffs.FilStorage{
 				ProposalCid:     di.ProposalCid,
-				Duration:        duration,
-				Miner:           d.Config.Miner,
+				Duration:        int64(di.Duration),
```

Comment:
Notice how `Duration` change its meaning.
Before we said that Duration was what the client specified in its deal proposal, that was wrong.
`Duration` should be (and is now) what the confirmed-active deal say it is.
This was an important fix since the "Start+Duration" consideration for deal renewals.